### PR TITLE
JCR-2415 :Allow to clean swap files after re-indexing data

### PR DESCRIPTION
**Pull request by aboughzela**
Swap file is used as temporary location where the binary value will be spooled if no value storage is configured and the  max-buffer-size is exceeded.
 * It's possible to force the clean swap files thanks to the System property exo.jcr.spoolConfig.force.clean.swap. The default value of this parameter is false (force clean disabled).
*  If force clean is enabled, It's possible to define the live time (in seconds) of swap files thanks to the System property exo.jcr.spoolConfig.swap.live.time.  The default value of this parameter is -1 (disabled).
* If force clean swap is disabled, swap files is removed until cached binary property data is evicted or reload data container.

### CODE REVIEW

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java [hunk 1]
```diff
@@ -1054,6 +1059,50 @@ private void reindex(boolean doReindexing, boolean doForceReindexing, boolean do
...
+                     {
+                        SwapFile swap = SwapFile.get(directory, file.getName());
+                        if(swap != null)
+                        {
+                           swap.delete();
+                        }
+                     }
+                     catch (IOException e)
+                     {
+                        if(log.isDebugEnabled())
```
Comment (thomasdelhomenie) [inline exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java]: This condition should be removed. The log must be displayed whenever something goes wrong and it should be an error, not a warn IMO.

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java [hunk 2]
```diff
@@ -1054,6 +1059,50 @@ private void reindex(boolean doReindexing, boolean doForceReindexing, boolean do
...
+                        {
+                           log.warn("unable to delete swap files after re-index " + WorkspaceName);
```
Comment (thomasdelhomenie) [inline exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java]: Do not swallow the exception

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/dataflow/SpoolConfig.java [hunk 3]
```diff
@@ -37,8 +38,43 @@
...
+      if (forceClean)
+      {
+         try
+         {
+            liveTime = Integer.parseInt(System.getProperty(FORCE_CLEAN_SWAP_LIVE_TIME));
+
+         }
+         catch (NumberFormatException nex)
+         {
+            liveTime = -1;
```
Comment (thomasdelhomenie) [inline exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/dataflow/SpoolConfig.java]: We should add a log here (warn) to let the customer know that its setting is not correct.

File: exo.jcr.docs/exo.jcr.docs.developer/en/src/main/docbook/en-US/modules/jcr/configuration/workspace-persistence-storage.xml [hunk 4]
```diff
@@ -134,6 +134,15 @@
     url="http://en.wikipedia.org/wiki/Bloom_filter&quot; &gt;http://en.wikipedia.org/wiki/Bloom_filter">here</ulink>.</para>
   </note>
 
+  <note>
+   <para>
+     It's possible to force the clean swap files thanks to the System property exo.jcr.spoolConfig.force.clean.swap.
+     The default value of this parameter is false (force clean disabled).
+     If force clean is enabled, It's possible to define the live time (in seconds) of swap files thanks to the System property exo.jcr.spoolConfig.swap.live.time.
+     The default value of this parameter is -1 (disabled).
```
Comment (thomasdelhomenie) [inline exo.jcr.docs/exo.jcr.docs.developer/en/src/main/docbook/en-US/modules/jcr/configuration/workspace-persistence-storage.xml]: We should add :
- what happens if we do not enabled this property
- what is the role of these swap files
- what are the benefits/drawbacks of setting a low/high value for exo.jcr.spoolConfig.swap.live.time

If will help the customer to decide which values to set.

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java (hunk 1 above)
Comment (aboughzela) [reply exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java]: change to error level, the error occur if the file doesn't existing any more when we try to remove it
(removed by another Thread before), So the log it's displayed only for debug level

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java (hunk 2 above)
Comment (aboughzela) [reply exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/core/query/lucene/SearchIndex.java]: it's ok to change to error level, the error occur if the file doesn't existing any more when we try to remove it
(removed by another Thread before), So the log it's displayed only for debug level

File: exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/dataflow/SpoolConfig.java (hunk 3 above)
Comment (aboughzela) [reply exo.jcr.component.core/src/main/java/org/exoplatform/services/jcr/impl/dataflow/SpoolConfig.java]: ok

File: exo.jcr.docs/exo.jcr.docs.developer/en/src/main/docbook/en-US/modules/jcr/configuration/workspace-persistence-storage.xml (hunk 4 above)
Comment (aboughzela) [reply exo.jcr.docs/exo.jcr.docs.developer/en/src/main/docbook/en-US/modules/jcr/configuration/workspace-persistence-storage.xml]: Description updated

**Pull request closed without merging**